# <center>Capstone Project 1 - 'Credit Card Approval Prediction' </center>
### <center>Notebook 3 - SQL Queries </center>

In [1]:
import duckdb
import pandas as pd
# Reading the cleaned credit data from a CSV file saved in the notebook 1
Credit_data = pd.read_csv('Credit_data_cleaned_SQL.csv')

In [2]:
Credit_data.columns

Index(['Ind_ID', 'Gender', 'Car_Owner', 'Propert_Owner', 'Children',
       'Annual_income', 'Type_Income', 'Education', 'Marital_status',
       'Housing_type', 'Family_Members', 'Label', 'Age', 'Employed_Years',
       'Age_Category'],
      dtype='object')

In [3]:
conn = duckdb.connect()
conn.register('Credit_data',Credit_data)

### 1. Group the customers based on their income type and find the average of their annual income.

In [4]:
query = """ Select Type_Income, round(avg(Annual_income),2) as Average_Annual_income 
            from Credit_data
            group by Type_Income
        """
conn.execute(query).fetchdf()

Type_Income  Average_Annual_income
0             Pensioner              135000.00
1               Working              173055.61
2         State servant              192957.83
3  Commercial associate              197777.27

### 2. Find the female owners of cars and property.

In [5]:
query = """ Select Ind_ID, Gender, Car_Owner, Propert_Owner
            From Credit_data
            Where Gender = 'F' and Car_Owner = 'Y' and Propert_Owner = 'Y'
            Group By Ind_ID, Gender, Car_Owner, Propert_Owner
        """
print("ID's of the female owners of cars and property")
conn.execute(query).fetchdf()

ID's of the female owners of cars and property


Ind_ID Gender Car_Owner Propert_Owner
0    5053254      F         Y             Y
1    5125716      F         Y             Y
2    5105614      F         Y             Y
3    5090861      F         Y             Y
4    5053003      F         Y             Y
..       ...    ...       ...           ...
119  5069466      F         Y             Y
120  5037202      F         Y             Y
121  5149296      F         Y             Y
122  5091708      F         Y             Y
123  5048458      F         Y             Y

[124 rows x 4 columns]

### 3. Find the male customers who are staying with their families.

In [6]:
query1 = """ Select Gender, Housing_type, Count(*) as Total_Count 
            From Credit_data
            Where Gender = 'M' and Housing_type = 'With parents'
            Group By Gender, Housing_type
        """
print("Total number of male customers who are staying with their families")
conn.execute(query1).fetchdf()

Total number of male customers who are staying with their families


Gender  Housing_type  Total_Count
0      M  With parents           32

In [7]:
query2 = """ Select Ind_ID, Gender, Housing_type
            From Credit_data
            Where Gender = 'M' and Housing_type = 'With parents'
            Group By Ind_ID, Gender, Housing_type
        """
print("ID's of male customers who are staying with their families")
conn.execute(query2).fetchdf()

ID's of male customers who are staying with their families


Ind_ID Gender  Housing_type
0   5050729      M  With parents
1   5054407      M  With parents
2   5143025      M  With parents
3   5028383      M  With parents
4   5150038      M  With parents
5   5069007      M  With parents
6   5143019      M  With parents
7   5058267      M  With parents
8   5037125      M  With parents
9   5125542      M  With parents
10  5033496      M  With parents
11  5009269      M  With parents
12  5079167      M  With parents
13  5145694      M  With parents
14  5146429      M  With parents
15  5143560      M  With parents
16  5142781      M  With parents
17  5024592      M  With parents
18  5066915      M  With parents
19  5010202      M  With parents
20  5126311      M  With parents
21  5028612      M  With parents
22  5143573      M  With parents
23  5038751      M  With parents
24  5010203      M  With parents
25  5113302      M  With parents
26  5079166      M  With parents
27  5126751      M  With parents
28  5054414      M  With parents
29  5045537      M  With parents
30  5079168      M  With parents
31  5068648      M  With parents

### 4. Please list the top five people having the highest income.

In [8]:
query = """ Select *
            From Credit_data
            Order by Annual_Income desc
            Limit 5
        """
print("Details of  the top five people having the highest income")
conn.execute(query).fetchdf()

Details of  the top five people having the highest income


Ind_ID Gender Car_Owner Propert_Owner  Children  Annual_income  \
0  5044935      M         N             Y         2       360000.0   
1  5146463      M         Y             N         0       360000.0   
2  5066597      F         N             N         1       360000.0   
3  5028274      F         Y             N         2       360000.0   
4  5066817      F         Y             N         1       360000.0   

            Type_Income            Education Marital_status Housing_type  \
0         State servant     Higher education        Married    Apartment   
1         State servant     Higher education        Married    Apartment   
2  Commercial associate     Higher education        Married    Apartment   
3               Working  Secondary Education        Married    Apartment   
4               Working     Higher education          Widow    Apartment   

   Family_Members     Label  Age  Employed_Years Age_Category  
0               4  Approved   43           15.44        Adult  
1               2  Approved   27            5.26        Young  
2               3  Approved   40            5.26        Adult  
3               4  Approved   34            6.06        Young  
4               2  Approved   45           11.01        Adult

### 5. How many married people are having bad credit?

In [9]:
query = """ Select Marital_status, Label, Count(*) as Bad_Credit
            From Credit_data
            Where Marital_status = 'Married' and Label = 'Rejected'
            Group By Marital_status, Label
        """
conn.execute(query).fetchdf()

Marital_status     Label  Bad_Credit
0        Married  Rejected          70

- Looking at the results above 70 Married People have bad credit

### 6. What is the highest education level and what is the total count?

In [10]:
query = """ Select Education, Count(*) as Total_Count
            From Credit_data
            Group By Education
            Order By Education
        """
conn.execute(query).fetchdf()

Education  Total_Count
0      Academic degree            1
1     Higher education          305
2      Lower secondary           14
3  Secondary Education          727

- Looking at the results above Higher education is the highest education level with 305 total count.

### 7. Between married males and females, who is having more bad credit? 

In [11]:
# SQL query to determine who, between married males and females, has more bad credit
query = """ Select Gender, Marital_Status, Label, Count(*) as Bad_Credit
            From Credit_data
            Where Marital_Status = 'Married' and Label = 'Rejected'
            Group By Gender, Marital_Status, Label
        """
conn.execute(query).fetchdf()

Gender Marital_status     Label  Bad_Credit
0      F        Married  Rejected          33
1      M        Married  Rejected          37

- Looking at the results above More Married Male have bad credit